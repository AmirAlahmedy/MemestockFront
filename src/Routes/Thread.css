.threadContainer{
    width:600px;
}
.threadSubreddit{
    font-size: 12px;
    font-weight: 700;
    color: rgb(215, 218, 220);
    display: inline;
    line-height: 20px;
    vertical-align: baseline;
    text-decoration: none;
}
.threadSubreddit:hover{
    text-decoration: underline;
}
.threadTitle{
    display: inline;
    color: rgb(215, 218, 220);
    overflow-wrap: break-word;
    font-size: 20px;
    font-weight: 500;
    line-height: 24px;
    padding-right: 5px;
}
.threadUsername{
    
    font-size: 12px;
    font-weight: 700;
    display: inline;
    line-height: 20px;
    vertical-align: baseline;
    text-decoration: none;
    color: rgb(111, 112, 113);
}
.threadUsername:hover{
    text-decoration: underline;
}
.threadContent{
    display: block;
    margin-block-start: 1em;
    margin-block-end: 1em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    color: rgb(215, 218, 220);
    word-break: break-word;
}
.threadComments { 
    display: inline-block;
    line-height: 1;
    text-transform: capitalize;
    vertical-align: middle;
}
