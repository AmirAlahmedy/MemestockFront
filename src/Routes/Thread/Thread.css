.threadContainer{
    width:600px;
}
.threadSubreddit{
    font-size: 12px;
    font-weight: 700;
    color: rgb(215, 218, 220);
    display: inline;
    line-height: 20px;
    vertical-align: baseline;
    text-decoration: none;
    float:left;
    margin-left:10px;
    margin-top:10px;
}
.threadSubreddit:hover{
    text-decoration: underline;
}
.threadTitle{
  display: block;
   color: rgb(233, 172, 92);
    overflow-wrap: break-word;
    word-break: break-word;
    text-align: left;
    font-size: large;
    font-weight: bold;
    margin-bottom: 1px;
    line-height: 24px;
    padding-right: 5px;
    padding-left:20px;
    margin-left: -10px;

}
.threadUsername{
    font-size: 12px;
    font-weight: 700;
    display: inline-block;
    word-break: break-word;
    line-height: 20px;
    vertical-align: baseline;
    text-decoration: none;
    color: rgb(111, 112, 113);
    float: left;
    margin-left:10px;
    margin-top:3px;
}
.threadUsername:hover{
    text-decoration: underline;
}
.threadContent{
    display: block;
    font-size:16px;
    margin-block-start: 1em;
    margin-block-end: 1em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    color: rgb(215, 218, 220);
    word-break: break-word;
    text-align: left;
    margin-left:10px;
    margin-top:10px;
}
.incrementVotes{
    margin-left: 40px;
    font-size:20px;
    cursor: pointer;
    float:left;
    background-color: rgb(87, 67, 67);
    border-radius: 10px;
    content: "\21E7";
   

}
.incrementVotes:hover {
    background-color: #59595a52;
    border-radius: 7px;
    
}
.threadUpvotes{

float:left;
font-size:19px;
margin-left: 20px;
margin-bottom: 5px

}

.decrementVotes{
    cursor: pointer;
    font-size:20px;
    float:left;
    margin-left: 25px;
    background-color: rgb(87, 67, 67);
    border-radius: 5px;
    content: "\21E9";
}
.decrementVotes:hover {
    background-color: #59595a52;
    border-radius: 10px;
}


.threadComments { 
    
    display: inline-block;
   
    text-align: middle;
    color: #f2f2f2; 
    font-size: 20px;
    cursor: pointer;

    width: 160px;
    height: 30px;
    background: rgb(87, 67, 67);
    position: relative;
    -moz-border-radius: 10px;
    -webkit-border-radius: 10px;
    border-radius: 10px;
    margin-bottom: 5px;
}
.threadComments:before {
    content: "";
    position: absolute;
    right: 100%;
    top: 8px;
    width: 0;
    height: 0;
    border-top: 6px solid rgb(29, 27, 27);
    border-right: 14px solid rgb(87, 67, 67);
    border-bottom: 6px solid rgb(29,27,27);
   
}