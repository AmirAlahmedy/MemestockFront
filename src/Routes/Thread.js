import React, { Component } from 'react';
import './Thread.css';
import { NavLink} from 'react-router-dom';
import Subreddit from './Subreddit';

class Thread extends Component {
    state = { 
        username:'GiantSteps_',
        subreddit:'jazztheory',
        title:'Modal Harmony Interchange', 
        content:'Modal Harmony is easier explained than played',
        comments:[
            {
                username:'WreakingHavoc',
                comment:'I rememeber when I had something to say!'
            },
            {
                username:'CluelessBastard',
                comment:'Oh this is so relatable'
            }
        ],
        upvotes:0
    }
    render(){
        return(
           <div>
               <div className="threadContainer">
                <NavLink className="threadSubreddit"  to='/r/'>r/{this.state.subreddit}</NavLink>
                <div className="threadUsername">
                    Posted by u/{this.state.username}
                </div>
                <div className="threadTitle"><h2>{this.state.title}</h2></div>
                <p className="threadContent">{this.state.content}</p>
                <div className="threadComments">
                    {this.state.comments.length} Comments
                </div>
                <div className="threadUpvotes">{this.state.upvotes}</div>
               </div>
           </div> 
        );
    }
}

export default Thread;