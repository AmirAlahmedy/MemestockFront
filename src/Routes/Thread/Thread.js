import React, { Component } from 'react';
import './Thread.css';
import { Link } from 'react-router-dom';
import '../../Sass/styles.scss';

export default class Thread extends Component {

    state = {
        username: '',
        subreddit: '',
        title: '',
        content: '',
        comments: [],
        upvotes: 0,
        date:'',
        enableUp: false,
        enableDown: false
    }
    // mocks = data.threads[2];

    

    /**
     * Handles upvotes increments.
     * @function handleIncrement
     */
    handleIncrement = () => {


        if (this.state.enableUp === true && this.state.enableDown === false) {
            this.setState({ upvotes: this.state.upvotes - 1 });
            this.setState({ enableUp: false });

        }
        else if (this.state.enableUp === false && this.state.enableDown === false) {
            this.setState({ upvotes: this.state.upvotes + 1 });
            this.setState({ enableUp: true });

        }
        else if (this.state.enableDown === true && this.state.enableUp === false) {

            this.setState({ enableDown: false });
            this.setState({ enableUp: true });
            this.setState({ upvotes: this.state.upvotes + 2 });
        }

    };

    /**
     * Handles downvotes.
     * @function handledecrement
     */
    handledecrement = () => {

        if (this.state.enableDown === true && this.state.enableUp === false) {
            this.setState({ upvotes: this.state.upvotes + 1 });
            this.setState({ enableDown: false });
        }
        else if (this.state.enableDown === false && this.state.enableUp === false) {
            this.setState({ upvotes: this.state.upvotes - 1 });
            this.setState({ enableDown: true });
        }
        else if (this.state.enableUp === true && this.state.enableDown === false) {
            this.setState({ enableDown: true });
            this.setState({ enableUp: false });
            this.setState({ upvotes: this.state.upvotes - 2 });
        }

    };

    goTo(link){
        return function(){
            window.location.href = link;
        } 
    }

    render() {
        let thrdWrapper = this.props.view ? 'threadWrapper':'classicThreadWrapper';
        let thrdCont = this.props.view ? "threadContainer":'classicThreadContainer';
        let thrdContent = this.props.view ? 'threadContent':'classicThreadContent';
        return (
        
            <div /*className={thrdWrapper}*/>
                <div className={thrdCont}>
                    <div class="threadLinks">
                        <span onClick={this.goTo(`/r/${this.props.subreddit}`)} className="threadSubreddit"> r/{this.props.subreddit}</span>
                        .
                        <span onClick={this.goTo(`/user/${this.props.username}`)}className="posted-by">   Posted by u/{this.props.username} </span>
                    </div>
                    <br></br>
                    <div className="threadTitle">{this.props.title}</div>

                    <p className={thrdContent}>{this.props.content}</p>

                    <button type="button" onClick={this.handleIncrement} className="incrementVotes"><i class="fas fa-angle-up"></i></button>

                    <div className="threadUpvotes">{this.state.upvotes}</div>

                    <button type="button" onClick={this.handledecrement} className="decrementVotes"><i class="fas fa-angle-down"></i></button>
                    <span onClick={this.goTo(`/thread/${this.props.id}?srName=${this.props.subreddit}`)} className="threadComments">
                        {window.location.href.includes("/thread/") ? "" : "View"}
                    </span>


                </div>
            </div>
        );
    }
}
