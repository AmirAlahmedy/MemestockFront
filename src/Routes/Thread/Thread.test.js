import React from "react";
import { create } from "react-test-renderer";
import Thread from "./Thread";


describe("Thread component snapshot", () => {
    test("it matches the snapshot", () => {
      const component = create(<Thread />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
  

  it("Shows Thread component meta", async () => {
    const component = create(<Thread />);
    const instance = component.getInstance();
    await instance.componentDidMount();
    console.log(instance.state); 
  });

  it("Increase votes by one", async () => {
    const component = create(<Thread />);
    const instance = component.getInstance();
    await instance.handleIncrement();
    console.log(instance.state); 
  });

  it("decrease votes by one", async () => {
    const component = create(<Thread />);
    const instance = component.getInstance();
    await instance.handledecrement();
    console.log(instance.state); 
  });
  
  
  
  