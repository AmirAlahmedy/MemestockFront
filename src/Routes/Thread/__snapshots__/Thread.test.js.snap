// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thread component snapshot it matches the snapshot 1`] = `
<div>
  <div
    className="classicThreadContainer"
  >
    <div
      class="threadLinks"
    >
      <span
        className="threadSubreddit"
        onClick={[Function]}
      >
         r/
      </span>
      .
      <span
        className="posted-by"
        onClick={[Function]}
      >
           Posted by u/
         
      </span>
    </div>
    <br />
    <div
      className="threadTitle"
    />
    <p
      className="classicThreadContent"
    />
    <button
      className="incrementVotes"
      onClick={[Function]}
      type="button"
    >
      <i
        class="fas fa-angle-up"
      />
    </button>
    <div
      className="threadUpvotes"
    />
    <button
      className="decrementVotes"
      onClick={[Function]}
      type="button"
    >
      <i
        class="fas fa-angle-down"
      />
    </button>
    <span
      className="threadComments"
      onClick={[Function]}
    >
      View
    </span>
  </div>
</div>
`;
